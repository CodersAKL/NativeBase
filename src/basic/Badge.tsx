import React, { Component, createRef } from 'react';
import { View, ViewStyle, StyleProp } from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';
type BadgeProps = {
  style?: StyleProp<ViewStyle>;
};

class Badge extends Component<BadgeProps, {}> {
  _root = createRef<View>();
  render() {
    return (
      <View ref={this._root} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}
const StyledBadge = connectStyle('NativeBase.Badge', {}, mapPropsToStyleNames)(Badge);

export { StyledBadge as Badge };
