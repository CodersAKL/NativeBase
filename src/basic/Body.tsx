import React, { Component, createRef } from 'react';
import { View, StyleProp, ViewStyle } from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';
type BodyProps = {
  style?: StyleProp<ViewStyle>;
};

class Body extends Component<BodyProps, {}> {
  _root = createRef<View>();
  render() {
    return <View ref={this._root} {...this.props} />;
  }
}
const StyledBody = connectStyle('NativeBase.Body', {}, mapPropsToStyleNames)(Body);

export { StyledBody as Body };
