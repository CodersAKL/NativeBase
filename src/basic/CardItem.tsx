import React, { Component } from 'react';
import { TouchableOpacity, View, StyleProp, ViewStyle } from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';
type CardItemProps = {
  style?: StyleProp<ViewStyle>;
  header?: boolean;
  cardBody?: boolean;
  footer?: boolean;
  button?: boolean;
};

class CardItem extends Component<CardItemProps, {}> {
  _root = React.createRef<View & TouchableOpacity>();
  render() {
    if (this.props.button) {
      return (
        <TouchableOpacity ref={this._root} activeOpacity={0.2} {...this.props}>
          {this.props.children}
        </TouchableOpacity>
      );
    }

    return (
      <View ref={this._root} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}
const StyledCardItem = connectStyle('NativeBase.CardItem', {}, mapPropsToStyleNames)(CardItem);

export { StyledCardItem as CardItem };
