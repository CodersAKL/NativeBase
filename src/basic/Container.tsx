import React, { Component } from 'react';
import { View, StyleProp, ViewStyle } from 'react-native';
import { connectStyle } from 'native-base-shoutem-theme';

import mapPropsToStyleNames from '../utils/mapPropsToStyleNames';
type ContainerProps = {
  style?: StyleProp<ViewStyle>;
};

class ContainerComponent extends Component<ContainerProps, {}> {
  _root = React.createRef<View>();
  render() {
    return (
      <View ref={this._root} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

export const Container = connectStyle('NativeBase.Container', {}, mapPropsToStyleNames)(ContainerComponent);
