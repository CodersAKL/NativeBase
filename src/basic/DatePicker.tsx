import React from 'react';
import {
  Modal,
  View,
  Platform,
  DatePickerIOS,
  DatePickerAndroid,
  DatePickerAndroidOpenOptions,
  ModalBaseProps,
  TextStyle,
  DatePickerIOSProps
} from 'react-native';

import variable from '../theme/variables/platform';
import { PLATFORM } from '../theme/variables/commonColor';
import { Text } from './Text';
type DatePickerProps = {
  disabled?: any;
  defaultDate?: Date;
  placeHolderText?: string;
  onDateChange: (date: Date) => void;
  minimumDate?: Date;
  maximumDate?: Date;
  androidMode?: DatePickerAndroidOpenOptions['mode'];
  formatChosenDate?: (date: Date) => void;
  locale?: string;
  animationType?: ModalBaseProps['animationType'];
  modalTransparent?: ModalBaseProps['transparent'];
  placeHolderTextStyle?: TextStyle;
  textStyle?: TextStyle;
  timeZoneOffsetInMinutes?: DatePickerIOSProps['timeZoneOffsetInMinutes'];
};
type DatePickerState = {
  chosenDate?: Date;
  defaultDate: Date;
  modalVisible: boolean;
};

export class DatePicker extends React.Component<DatePickerProps, DatePickerState> {
  constructor(props: DatePickerProps) {
    super(props);
    this.state = {
      modalVisible: false,
      defaultDate: props.defaultDate ? props.defaultDate : new Date(),
      chosenDate: !props.placeHolderText && props.defaultDate ? props.defaultDate : undefined
    };
  }

  setDate(date: Date) {
    this.setState({ chosenDate: new Date(date) });
    if (this.props.onDateChange) {
      this.props.onDateChange(date);
    }
  }
  showDatePicker = () => {
    if (Platform.OS === PLATFORM.ANDROID) {
      this.openAndroidDatePicker();
    } else {
      this.setState({ modalVisible: true });
    }
  };
  async openAndroidDatePicker() {
    try {
      const newDate = await DatePickerAndroid.open({
        date: this.state.chosenDate ? this.state.chosenDate : this.state.defaultDate,
        minDate: this.props.minimumDate,
        maxDate: this.props.maximumDate,
        mode: this.props.androidMode
      });

      if (newDate.action === 'dateSetAction') {
        const { year, month, day } = newDate;
        const selectedDate = new Date(year, month, day);

        this.setState({ chosenDate: selectedDate });
        this.props.onDateChange(selectedDate);
      }
    } catch ({ code, message }) {
      console.warn('Cannot open date picker', message);
    }
  }
  formatChosenDate(date: Date) {
    if (this.props.formatChosenDate) {
      return this.props.formatChosenDate(date);
    }

    return [date.getDate(), date.getMonth() + 1, date.getFullYear()].join('/');
  }
  render() {
    const {
      animationType,
      disabled,
      locale,
      maximumDate,
      minimumDate,
      modalTransparent,
      placeHolderText,
      placeHolderTextStyle,
      textStyle,
      timeZoneOffsetInMinutes
    } = this.props;
    const variables = this.context.theme ? this.context.theme['@@shoutem.theme/themeStyle'].variables : variable;

    return (
      <View>
        <View>
          <Text
            onPress={() => (!disabled ? this.showDatePicker() : undefined)}
            style={[
              {
                padding: variables.datePickerPadding,
                color: variables.datePickerTextColor
              },
              this.state.chosenDate ? textStyle : placeHolderTextStyle
            ]}
          >
            {this.state.chosenDate ? this.formatChosenDate(this.state.chosenDate) : placeHolderText || 'Select Date'}
          </Text>
          <View>
            <Modal
              supportedOrientations={['portrait', 'landscape']}
              animationType={animationType}
              transparent={modalTransparent} // from api
              visible={this.state.modalVisible}
              onRequestClose={() => {}}
            >
              <Text
                onPress={() => this.setState({ modalVisible: false })}
                style={{
                  backgroundColor: variables.datePickerBg,
                  flex: variables.datePickerFlex
                }}
              />
              <DatePickerIOS
                date={this.state.chosenDate ? this.state.chosenDate : this.state.defaultDate}
                onDateChange={date => this.setDate(date)}
                minimumDate={minimumDate}
                maximumDate={maximumDate}
                mode="date"
                locale={locale}
                timeZoneOffsetInMinutes={timeZoneOffsetInMinutes}
              />
            </Modal>
          </View>
        </View>
      </View>
    );
  }
}
