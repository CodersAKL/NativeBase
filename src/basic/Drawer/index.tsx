import React, { Component, createRef } from 'react';
import RNDrawer from 'react-native-drawer';

export default class Drawer extends Component<{}, {}> {
  _root = createRef<RNDrawer>();
  static defaultProps = {
    type: 'overlay',
    tapToClose: true,
    openDrawerOffset: 0.2,
    panCloseMask: 0.2,
    closedDrawerOffset: 0,
    styles: {
      drawer: {
        shadowColor: '#000000',
        shadowOpacity: 0,
        shadowRadius: 0,
        elevation: 5,
      },
      mainOverlay: {
        opacity: 0,
        backgroundColor: 'rgba(0, 0, 0, 0.8)',
        elevation: 0,
      },
    },
    tweenHandler: (ratio: number) => ({
      mainOverlay: { opacity: ratio / 2 },
    }),
  };

  render() {
    return <RNDrawer ref={this._root} {...this.props} />;
  }
}
