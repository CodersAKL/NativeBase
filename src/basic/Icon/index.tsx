import React, { Component, createRef } from 'react';
import { Platform, StyleProp, ViewStyle } from 'react-native';
import { object } from 'prop-types';
import { connectStyle } from 'native-base-shoutem-theme';

import variable from '../../theme/variables/platform';
import mapPropsToStyleNames from '../../utils/mapPropsToStyleNames';
import { IconNB } from '../IconNB';
import ic from './NBIcons.json';

export type IconName = keyof typeof ic;

type IconProps = {
  style?: StyleProp<ViewStyle>;
  name: IconName;
  ios?: string;
  android?: string;
  active?: boolean;
  type?: string;
};

export class IconComponent extends Component<IconProps, {}> {
  _root = createRef<typeof IconNB>();
  static contextTypes = {
    theme: object,
  };
  getName() {
    const variables = this.context.theme ? this.context.theme['@@shoutem.theme/themeStyle'].variables : variable;
    const platformStyle = variables.platformStyle;
    const platform = variables.platform;

    if ((this.props.type || variables.iconFamily) === 'Ionicons') {
      if (typeof ic[this.props.name] !== 'object') {
        return this.props.name;
      } else if (typeof ic[this.props.name] === 'object') {
        let name;

        if (platform === 'ios' && platformStyle !== 'material') {
          name = this.props.active ? ic[this.props.name].ios.active : ic[this.props.name].ios.default;
        } else {
          name = this.props.active ? ic[this.props.name].android.active : ic[this.props.name].android.default;
        }

        return name;
      }
    } else {
      return this.props.name;
    }

    return null;
  }
  getIconName() {
    if (Platform.OS === 'ios') {
      if (this.props.ios) {
        return this.props.ios;
      }

      return this.props.active ? ic[this.props.name].ios.active : ic[this.props.name].ios.default;
    } else if (this.props.android) {
      return this.props.android;
    }

    return this.props.active ? ic[this.props.name].android.active : ic[this.props.name].android.default;
  }
  render() {
    if (this.props.ios && this.props.android) {
      return (
        <IconNB ref={this._root} {...this.props} name={Platform.OS === 'ios' ? this.props.ios : this.props.android} />
      );
    } else if (this.props.name && (this.props.android || this.props.ios)) {
      return <IconNB ref={this._root} {...this.props} name={this.getIconName()} />;
    }

    return <IconNB ref={this._root} {...this.props} name={this.getName()} />;
  }
}
const StyledIcon = connectStyle('NativeBase.Icon', {}, mapPropsToStyleNames)(IconComponent);

export { StyledIcon as Icon };
