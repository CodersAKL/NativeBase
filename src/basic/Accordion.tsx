import React from 'react';
import {
  Animated,
  TouchableWithoutFeedback,
  FlatList,
  StyleSheet,
  View,
  ViewStyle,
  TextStyle,
  StyleProp,
} from 'react-native';

import variable from '../theme/variables/platform';
import { Text } from './Text';
import { Icon, IconName } from './Icon';
type AccordionSubItemState = {
  fadeAnim: Animated.Value;
};
type AccordionState = {
  selected: undefined | any;
};

interface DefaultHeaderProps {
  expanded: boolean;
  expandedIcon: IconName;
  expandedIconStyle: StyleProp<ViewStyle>;
  headerStyle: StyleProp<ViewStyle>;
  icon: IconName;
  iconStyle: StyleProp<ViewStyle>;
  title: string;
}

class DefaultHeader extends React.Component<DefaultHeaderProps, {}> {
  render() {
    const { expanded, expandedIcon, expandedIconStyle, headerStyle, icon, iconStyle, title } = this.props;
    const variables = this.context.theme ? this.context.theme['@@shoutem.theme/themeStyle'].variables : variable;

    return (
      <View style={[styles.defaultHeader, headerStyle || { backgroundColor: variables.headerStyle }]}>
        <Text> {title}</Text>
        <Icon
          style={[
            { fontSize: variables.accordionIconFontSize },
            expanded
              ? expandedIcon && expandedIconStyle
                ? expandedIconStyle
                : { color: variables.expandedIconStyle }
              : icon && iconStyle
              ? iconStyle
              : { color: variables.iconStyle },
          ]}
          name={expanded ? expandedIcon || 'ios-arrow-up' : icon || 'ios-arrow-down'}
        />
      </View>
    );
  }
}

interface DefaultContentProps {
  contentStyle: TextStyle;
  content: string;
}

class DefaultContent extends React.Component<DefaultContentProps, {}> {
  render() {
    const { content, contentStyle } = this.props;
    const variables = this.context.theme ? this.context.theme['@@shoutem.theme/themeStyle'].variables : variable;

    return (
      <Text
        style={[
          { padding: variable.accordionContentPadding },
          contentStyle || { backgroundColor: variables.contentStyle },
        ]}
      >
        {content}
      </Text>
    );
  }
}

interface AccordionSubItemProps {
  style?: ViewStyle;
}

class AccordionSubItem extends React.Component<AccordionSubItemProps, AccordionSubItemState> {
  state = {
    fadeAnim: new Animated.Value(0.3),
  };
  componentDidMount() {
    Animated.timing(this.state.fadeAnim, {
      toValue: 1,
      duration: 500,
      useNativeDriver: true,
    }).start();
  }
  render() {
    const { children, style } = this.props;
    const { fadeAnim } = this.state;

    return <Animated.View style={{ ...style, opacity: fadeAnim }}>{children}</Animated.View>;
  }
}

interface AccordionItemProps {
  contentStyle: ViewStyle;
  expanded: boolean;
  expandedIcon: IconName;
  expandedIconStyle: any;
  headerStyle: any;
  icon: IconName;
  iconStyle: any;
  index: number;
  item: any;
  onAccordionClose: any;
  onAccordionOpen: any;
  renderContent: any;
  renderHeader: any;
  setSelected: (i: number) => void;
}

class AccordionItem extends React.Component<AccordionItemProps, {}> {
  render() {
    const {
      contentStyle,
      expanded,
      expandedIcon,
      expandedIconStyle,
      headerStyle,
      icon,
      iconStyle,
      index,
      item,
      onAccordionClose,
      onAccordionOpen,
      renderContent,
      renderHeader,
      setSelected,
    } = this.props;

    return (
      <View>
        <TouchableWithoutFeedback
          onPress={() => {
            onAccordionOpen && !expanded && onAccordionOpen(item, index);
            onAccordionClose && expanded && onAccordionClose(item, index);
            setSelected(index);
          }}
        >
          <View>
            {renderHeader ? (
              renderHeader(item, expanded)
            ) : (
              <DefaultHeader
                expanded={expanded}
                expandedIcon={expandedIcon}
                expandedIconStyle={expandedIconStyle}
                headerStyle={headerStyle}
                icon={icon}
                iconStyle={iconStyle}
                title={item.title}
              />
            )}
          </View>
        </TouchableWithoutFeedback>
        {expanded ? (
          <AccordionSubItem>
            {renderContent ? (
              renderContent(item)
            ) : (
              <DefaultContent content={item.content} contentStyle={contentStyle} />
            )}
          </AccordionSubItem>
        ) : null}
      </View>
    );
  }
}

interface AccordionProps {
  expanded?: boolean;
  contentStyle?: ViewStyle;
  dataArray: any;
  expandedIcon?: IconName;
  expandedIconStyle?: any;
  headerStyle?: any;
  icon?: IconName;
  iconStyle?: any;
  onAccordionClose?: any;
  onAccordionOpen?: any;
  renderContent?: any;
  renderHeader?: any;
  style?: ViewStyle;
}

export class Accordion extends React.Component<AccordionProps, AccordionState> {
  constructor(props: Readonly<AccordionProps>) {
    super(props);
    this.state = {
      selected: props.expanded,
    };
  }
  setSelected(index: number) {
    if (this.state.selected === index) {
      this.setState({ selected: undefined });
    } else {
      this.setState({ selected: index });
    }
  }
  render() {
    const {
      contentStyle,
      dataArray,
      expandedIcon,
      expandedIconStyle,
      headerStyle,
      icon,
      iconStyle,
      onAccordionClose,
      onAccordionOpen,
      renderContent,
      renderHeader,
      style,
    } = this.props;
    const variables = this.context.theme ? this.context.theme['@@shoutem.theme/themeStyle'].variables : variable;

    return (
      <FlatList
        data={dataArray}
        extraData={this.state}
        style={[
          {
            borderColor: variables.accordionBorderColor,
            borderWidth: variables.borderWidth,
          },
          style,
        ]}
        keyExtractor={(item, index) => String(index)}
        renderItem={({ item, index }) => (
          <AccordionItem
            key={String(index)}
            contentStyle={contentStyle}
            expanded={this.state.selected === index}
            expandedIcon={expandedIcon}
            expandedIconStyle={expandedIconStyle}
            headerStyle={headerStyle}
            icon={icon}
            iconStyle={iconStyle}
            index={index}
            item={item}
            renderContent={renderContent}
            renderHeader={renderHeader}
            onAccordionOpen={onAccordionOpen}
            onAccordionClose={onAccordionClose}
            setSelected={i => this.setSelected(i)}
          />
        )}
        {...this.props}
      />
    );
  }
}
const styles = StyleSheet.create({
  defaultHeader: {
    flexDirection: 'row',
    padding: variable.accordionContentPadding,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});
