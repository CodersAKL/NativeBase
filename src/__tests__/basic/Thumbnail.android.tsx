import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Content } from '../../basic/Content';
import { Thumbnail } from '../../basic/Thumbnail';

const cover = require('../assets/drawer-cover.png');

it('renders default shape thumbnail', () => {
  const tree = renderer
    .create(
      <Content>
        <Thumbnail small source={cover} />
        <Thumbnail source={cover} />
        <Thumbnail large source={cover} />
      </Content>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
it('renders square shape thumbnail', () => {
  const tree = renderer
    .create(
      <Content>
        <Thumbnail square small source={cover} />
        <Thumbnail square source={cover} />
        <Thumbnail square large source={cover} />
      </Content>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
