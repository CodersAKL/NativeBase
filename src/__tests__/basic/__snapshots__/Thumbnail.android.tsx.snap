// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default shape thumbnail 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={Object {}}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Image
        small={true}
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        style={Object {}}
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        style={Object {}}
      />
      <Image
        large={true}
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        style={Object {}}
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders square shape thumbnail 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={Object {}}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Image
        small={true}
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        square={true}
        style={Object {}}
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        square={true}
        style={Object {}}
      />
      <Image
        large={true}
        source={
          Object {
            "testUri": "../../../src/__tests__/assets/drawer-cover.png",
          }
        }
        square={true}
        style={Object {}}
      />
    </View>
  </RCTScrollView>
</View>
`;
